File: Daily/PY/Leetcode832-翻转图像.py
from typing import List
class Solution:
    def flipAndInvertImage(self, A: List[List[int]]) -> List[List[int]]:
        m ,n = len(A),len(A[0])
        for i in range(m):
            start,end = 0,n - 1
            while start < end:
                if A[i][start] == A[i][end]:
                    A[i][start] ^= 1
                    A[i][end] ^= 1
                start += 1
                end -= 1
            if start == end:
                A[i][start] ^= 1
        return A

File: Daily/Rust/Leetcode832-翻转图像.rs
//01互换的方法 1 - a和 ^= 1
impl Solution {
    pub fn flip_and_invert_image(a: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let (m,n) = (a.len(),a[0].len());
        let mut a = a;
        for i in 0..m{
            let (mut start,mut end) = (0,n - 1);
            while start < end{
                if a[i][start] == a[i][end]{
                    a[i][start] ^= 1;
                    a[i][end] ^= 1;    
                }
                start += 1;
                end -= 1;
            }
            if start == end{
                a[i][start] ^= 1;
            }
        }       
        a
    }
}